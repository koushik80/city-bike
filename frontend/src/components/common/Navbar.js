import { Container } from '@mui/material'
import React from 'react'
import { Link, NavLink } from 'react-router-dom'
import logo from './../../image/c6.gif'
import './Navbar.scss'

const Navbar = () => {
  return (
    <Container>
      <nav className='navbar'>
        <div className='navbar--logo'>
          <Link to="/">
            <img src={logo} alt="" />
          </Link>
          <div className="wrapper">
            <h1>Ci†yßîke</h1>
          </div>
        </div>
        <div className='navbar--link-list'>
          <ul>
            <li>
              <NavLink to="/">Home</NavLink>
            </li>
            <li>
              <NavLink to="/journey">Journey</NavLink>
            </li>
            <li>
              <NavLink to="/station">Stations</NavLink>
            </li>
            <li>
              <NavLink to="/contact">Contact</NavLink>
            </li>
          </ul>
        </div>
      </nav>
    </Container>
  )
}

export default Navbar
