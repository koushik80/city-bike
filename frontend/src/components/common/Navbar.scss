.navbar {
  display: flex;
  align-items: center;
  justify-content: space-between;

  &--logo {
    padding: 0.2rem 0rem;
    cursor: pointer;
    img {
      width: 55px;
      height: 35px;
      object-fit: contain;
    }
  }

  &--link-list {
    ul {
      display: flex;
      gap: 16px;
      li {
        a {
          padding-bottom: 2px;
          &:hover {
            color: #5670ff;
            border-bottom: 5px solid #415cf7;
          }
        }
      }
    }
  }
}

.wrapper {
  text-align: center;
  h1 {
    padding: 0.1rem;
    color: #fff;
    font-size: 14px;
    font-weight: 900;
    font-family: "Josefin Sans", sans-serif;
    background: linear-gradient(to right,#BFB8DA 10%, #56c596 10%, #7BE495 10%, #96D7D6 20%, #5E72EB 20%, #4ADADE 20%, #C2C8C5 20%, #A8DEE0 80%, #FF9190 20%);
    background-size: auto auto;
    background-clip: border-box;
    background-size: 200% auto;
    color: #fff;
    background-clip: text;
    text-fill-color: transparent;
    -webkit-background-clip: text;
    -webkit-text-fill-color: transparent;
    animation: textclip 5s linear infinite;
    display: inline-block;
}
}

@keyframes textclip {
  to {
    background-position: 200% center;
  }
}

@media (max-width:600px) {

        &-logo{
            h1{
              font-size: 15px;
            }
        }
    }
    @media (max-width:850px) {

        &-logo{
            h1{
              font-size: 15px;
            }
        }
    }