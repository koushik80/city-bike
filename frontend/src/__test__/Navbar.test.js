import React from 'react'
import { BrowserRouter } from 'react-router-dom'
import renderer from 'react-test-renderer'
import Navbar from '../components/common/Navbar'


describe('Home Navbar', () => {

  it('should render correctly', () => {
    const wrapper = renderer.create(<BrowserRouter><Navbar /></BrowserRouter> )

    expect(wrapper).toMatchSnapshot()
  })
})