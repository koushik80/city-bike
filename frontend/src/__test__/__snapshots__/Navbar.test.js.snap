// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Navbar should render correctly 1`] = `
<div
  className="MuiContainer-root MuiContainer-maxWidthLg css-1oqqzyl-MuiContainer-root"
>
  <nav
    className="navbar"
  >
    <div
      className="navbar--logo"
    >
      <a
        href="/"
        onClick={[Function]}
      >
        <img
          alt=""
          src="c6.gif"
        />
      </a>
      <div
        className="wrapper"
      >
        <h1>
          Ci†yßîke
        </h1>
      </div>
    </div>
    <div
      className="navbar--link-list"
    >
      <ul>
        <li>
          <a
            aria-current="page"
            className="active"
            href="/"
            onClick={[Function]}
          >
            Home
          </a>
        </li>
        <li>
          <a
            className=""
            href="/journey"
            onClick={[Function]}
          >
            Journey
          </a>
        </li>
        <li>
          <a
            className=""
            href="/station"
            onClick={[Function]}
          >
            Stations
          </a>
        </li>
        <li>
          <a
            className=""
            href="/travelling"
            onClick={[Function]}
          >
            Travelling
          </a>
        </li>
        <li>
          <a
            className=""
            href="/contact"
            onClick={[Function]}
          >
            Contact
          </a>
        </li>
      </ul>
    </div>
  </nav>
</div>
`;
